drop-transfer/src/manager.rs

```diff
@@ -635,16 +646,33 @@ impl TransferManager {
         state.ensure_not_terminated()
     }
 
-    pub async fn outgoing_remove(&self, transfer_id: Uuid) -> crate::Result<()> {
+    /// Returns `None` if the cancel event should be suppressed
+    pub async fn outgoing_remove(
+        &self,
+        transfer_id: Uuid,
+    ) -> crate::Result<Option<Arc<OutgoingTransferEventTx>>> {
         debug!(self.logger, "Removing outgoing transfer: {}", transfer_id);
         let mut lock = self.outgoing.lock().await;
         if !lock.contains_key(&transfer_id) {
             return Err(crate::Error::BadTransfer);
         }
         self.storage.transfer_sync_clear(transfer_id).await;
-        lock.remove(&transfer_id);
+        if let Some(state) = lock.remove(&transfer_id) {
+            if !matches!(state.xfer_sync, sync::TransferState::Canceled) {
+                return Ok(Some(state.xfer_events));
+            }
+        };
```

Comment:
Is it possible for this to be evaluate to `False`?

Reply:
Yes, in case we canceled the transfer and it was not communicated to the peer yet.

Reply:
Hmm.. Could you explain what happens then? I am a bit confused about the flow here and wonder if we should somehow handle the case then or at least log it

Reply:
So there are two ways to cancel the transfer. Either you can do it or your peer.
1) If your peer cancels the transfer, then it will hit its `return Ok(Some(..))` line and return event tx for the calling code to emit the event
2) If you cancel the transfer, then it's not removed but its state is set to `Canceled`. This is because you still need the transfer around for communication with the peer until he acknowledges the transfer is canceled. Then you can call remove and it will not execute the code under `if !matches!()`.

There's one more option. Your peer canceled at the same time as you. Then effectively remove will be called twice and the second time it will not execute the code under `if let Some(..) = lock.remove()`